sktime/datasets/_dataset_downloader.py

```diff
@@ -0,0 +1,342 @@
+"""Utilities for downloading datasets."""
+
+__author__ = ["jgyasu"]
+
+__all__ = ["DatasetDownloader", "HuggingFaceDownloader", "URLDownloader"]
+
+import os
+import shutil
+import tempfile
+import warnings
+import zipfile
+from pathlib import Path
+from urllib.error import HTTPError, URLError
+from urllib.request import urlretrieve
+
+from sktime.utils.dependencies import _check_soft_dependencies
+
+
+class DatasetDownloadStrategy:
+    """Base class for dataset download strategies."""
+
+    def download(
+        self, dataset_name, download_path=None, force_download=False, **kwargs
+    ):
+        """Download a dataset using this strategy.
+
+        This method handles common boilerplate, then delegates to
+        _download for strategy-specific implementation.
+
+        Parameters
+        ----------
+        dataset_name : str
+            Name of the dataset to download.
+        download_path : str or Path, optional
+            Local directory where the dataset should be saved. If not specified,
+            defaults to a 'local_data' folder in the current working directory.
+        force_download : bool, default=False
+            Whether to force re-download the dataset even if it's cached locally.
+        **kwargs
+            Additional keyword arguments passed to the strategy implementation.
+
+        Returns
+        -------
+        Path
+            Path to the downloaded dataset folder.
+        """
+        if download_path is None:
+            download_path = Path.cwd() / "local_data"
+        else:
+            download_path = Path(download_path)
+
+        download_path.mkdir(parents=True, exist_ok=True)
+        local_dataset_path = download_path / dataset_name
+
+        if local_dataset_path.exists():
```

Comment:
I think this can be simplified:

```python
if not local_dataset_path.exists() or force_download:
    self._download(dataset_name, download_path, **kwargs)
return local_dataset_path
```